pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -653,18 +653,48 @@ fi
 # If we're not just building, then make the new configuration the boot
 # default and/or activate it now.
 if [[ "$action" = switch || "$action" = boot || "$action" = test || "$action" = dry-activate ]]; then
+    # Using systemd-run here to protect against PTY failures/network
+    # disconnections during rebuild.
+    # See: https://github.com/NixOS/nixpkgs/issues/39118
+    cmd=(
+        "systemd-run"
+        "-E" "LOCALE_ARCHIVE" # Will be set to new value early in switch-to-configuration script, but interpreter starts out with old value
+        "--collect"
+        "--no-ask-password"
+        "--pty"
+        "--quiet"
+        "--same-dir"
+        "--service-type=exec"
+        "--unit=nixos-rebuild-switch-to-configuration"
+        "--wait"
+    )
+    # Check if we have a working systemd-run. In chroot environments we may have
+    # a non-working systemd, so we fallback to not using systemd-run.
+    # You may also want to explicitly set NIXOS_IGNORE_SYSTEMD_RUN environment
+    # variable, since systemd-run runs inside an isolated environment and
+    # this may break some post-switch scripts. However keep in mind that this
+    # may be dangerous in remote access (e.g. SSH).
+    if [[ -n "$NIXOS_IGNORE_SYSTEMD_RUN" ]]; then
+        log "warning: skipping systemd-run since NIXOS_IGNORE_SYSTEMD_RUN is set. This environment variable will be ignored in the future"
+        cmd=()
+    elif ! targetHostCmd "${cmd[@]}" true &>/dev/null; then
```

Comment:
When using `--target`, this is an extra ssh roundtrip that could be avoided if we make `switch-to-configuration` responsible for re-execing itself in `systemd-run` - ie move all the new logic into that command (or a custom wrapper for it).

Reply:
I took a look at the `switch-configuration` and it seems that this is in Perl and I am not really comfortable in modifying it.

Is this extra round-trip a blocker for this PR? Or could we leave this as-is and change it later?